pkg/search/backendstore/store.go

```diff
@@ -62,12 +62,19 @@ func DeleteBackend(cluster string) {
 
 // GetBackend get backend store
 func GetBackend(cluster string) BackendStore {
+	if bs, ok := func() (BackendStore, bool) {
+		backendLock.RLock()
+		defer backendLock.RUnlock()
+
+		bs, ok := backends[cluster]
+		return bs, ok
+	}(); ok {
+		return bs
+	}
+
 	backendLock.Lock()
```

Comment:
is it necessary lock for NewDefaultBackend?

Reply:
unnecessary

Reply:
If there are other goroutines writing to the backends map, `GetBackend` should be locked.  but is it necessary to use read/write locks? please corrent me if not